### PR TITLE
catch localStorage exception

### PR DESCRIPTION
**Pull request by zorafuchs**

### CODE REVIEW

File: src/app/data/services/local_storage_service.ts [hunk 1]
```diff
@@ -14,7 +14,10 @@ class LocalStorageService {
     if (this.skipStoring(key)) return;
 
     this.setExpiryDate(key);
-    localStorage.setItem(key, this.serializer.stringify(hash));
+    try {
+      localStorage.setItem(key, this.serializer.stringify(hash));
+    } catch (error) {
```
Comment (lukaselmer) [inline src/app/data/services/local_storage_service.ts]: we should log the actual error here:
```js
console && console.error && console.error(error);
```

Comment (zorafuchs) [inline src/app/data/services/local_storage_service.ts]: I inserted it. Could look over it and merge if its okey?

File: src/app/data/services/local_storage_service.spec.ts [hunk 2]
```diff
@@ -45,5 +45,24 @@ describe('LocalStorageService', function () {
...
+      spyOn(localStorage, 'setItem').and.callFake(() => {
+        throw error;
+      });
+      spyOn(console, 'error');
+
+      expect(function() {
+        service.put(key, hash1);
+      }).not.toThrow();
+
+      // it is not possible to mock the localStorage in firefox. Thats why we skip this test if it's run on firefox
```
Comment (lukaselmer) [inline src/app/data/services/local_storage_service.spec.ts]: 👍

File: src/app/data/services/local_storage_service.ts (hunk 1 above)
Comment (lukaselmer) [inline src/app/data/services/local_storage_service.ts]: nice 👍

**Pull request merged**